playbooks/files/rax-maas/plugins/octavia_check_quota_limits.py

```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/env python
+
+# Copyright 2017, Rackspace US, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import division
+
+import argparse
+import datetime
+
+import ipaddr
+from maas_common import get_auth_ref
+from maas_common import get_endpoint_url_for_service
+from maas_common import get_keystone_client
+from maas_common import metric
+from maas_common import metric_bool
+from maas_common import print_output
+from maas_common import status_err
+from maas_common import status_ok
+import requests
+
+
+def check(auth_ref, args):
+    keystone = get_keystone_client(auth_ref)
+    auth_token = keystone.auth_token
+
+    s = requests.Session()
+
+    s.headers.update(
+        {'Content-type': 'application/json',
+         'x-auth-token': auth_token})
+    try:
+        if args.tenant_id:
+            params = {'tenant_id': args.tenant_id,
+                      'project_id': args.tenant_id}
+        else:
+            params = {}
+
+        compute_endpoint = get_endpoint_url_for_service(
+                'compute', auth_ref, 'internal')
+
+        volume_endpoint = get_endpoint_url_for_service(
+                'volume', auth_ref, 'internal')
+
+        # network_endpoint = get_endpoint_url_for_service(
+        #         'volumev3', auth_ref, 'internal')
+
+
+        r = s.get('%s/limits' % compute_endpoint, params=params,
+            verify=False,
+            timeout=5)
+
+        if (r.status_code != 200):
+            raise Exception("Nova returned status code %s" % str(r.status_code))
+        nova = r.json()['limits']['absolute']
+
+        r = s.get('%s/limits' % volume_endpoint, params=params,
```

Comment:
This has the project ID in the URL: /v3/{project_id}/limits

Reply:
Yeah, not sure if that has bearing when ou pass in a project-id as a parameter...